### ISSUE TITLE
[CoE Starter Kit - BUG] COEDashboard Failed to load/refresh

### ISSUE THREAD

**Issue by Sinder88** [microsoft/coe-starter-kit#8240]
### Does this bug already exist in our backlog?

- [X] I have checked and confirm this is a new bug.

### Describe the issue

I have upgraded to the latest May version but when trying to open the COE Dashboard i get this error and am unable to fetch data:
![image](https://github.com/microsoft/coe-starter-kit/assets/109826106/dd745066-e767-4954-a043-fa4d80611d1f)

### What solution are you experiencing the issue with?

Core

### What solution version are you using?

4.31

### What app or flow are you having the issue with?

Production_CoEDashboard_May2024

### What method are you using to get inventory and telemetry?

Cloud flows

### Steps To Reproduce

1. Open the report
2. Insert the URL to my COE Environment
3. Refresh screen appears and i get the error message

AB#3194

**Comment by Jenefer-Monroe** [comment]
Sounds like you dont have access to the table which is odd. 
1. Can you please confirm, are you logged in to PBI as the user identity that setup the kit? 
1. Can you please share a screenshot like this so we can see what solutions you have installed in this environment? This way we'll know if there was an error in upgrade
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/8dd512cc-21fc-4c64-adab-0f64c29b9bf2)

**Comment by Sinder88** [comment]
So I initially installed the kit with AdminAccount1, i recently got a service account which now have full admin permissions like AdminAccount1. On Wednesday i upgraded the solution using the new service account based on the advise your team provided in April month Office Hours. So, i tried to open the dashboard using the new service account, yes. But perhaps its a permission issue based on the change of accounts in the Dataverse table itself. 

I had the same issue with Admin Planning where one of my staff added the data to Dataverse and my service account was unable to open the dashboard. I ended up deleting the Dataverse data and adding it with my service account and it worked. I hope there is a work around for the main COE as it definitely won't be so quick to replicate there. 

See screenshot requested:
![image](https://github.com/microsoft/coe-starter-kit/assets/109826106/d8483865-860d-4f5c-97bd-c30b20806a65)

**Comment by Jenefer-Monroe** [comment]
What SR does your new service account have in the CoE Envt? 
To use the PBI's they woudl want to have the Power Platform Admin SR
But if they are charged with the kit management (upgrades etc) then they should have the System Admin SR.

![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/c0a1c171-c986-4252-a3af-bb5885b219bd)
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/d84b6425-f9fc-4f21-b9ea-1ce3ecf4587c)

**Comment by Sinder88** [comment]
The SA is part of the PowerPLatform-Admin Security group which is part of the admin team in the environment which has the Power Platform Admin SR assigned.

**Comment by Sinder88** [comment]
So strange. I have tested using this new dashboard with 3 accounts:

1. Regular account (Not an admin - same error)
2. Cadmin Account (Original COE toolkit installer, Admin, Same error)
3. Service Account (Installed the latest upgrade, Admin, Same error)

The error seems to come from the AI Credit Usage table, so I changed the owner of the records to the Service account and i still get the same error. 

<img width="569" alt="image" src="https://github.com/microsoft/coe-starter-kit/assets/109826106/438a0538-fadb-486e-bbc8-9d6ccf52d08b">


<img width="658" alt="image" src="https://github.com/microsoft/coe-starter-kit/assets/109826106/b01d494c-5060-4f0f-8d8e-e92f8687a9ea">
